### PR TITLE
AGENT-998: add job to run node-joiner integration tests

### PR DESCRIPTION
**Pull request by andfasano**
This patch adds a new job to execute the node-joiner integration tests in the installer CI repo

### CODE REVIEW

**Comment by andfasano** [comment]
/hold

Requires https://github.com/openshift/installer/pull/9064

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml [hunk 1]
```diff
@@ -890,6 +890,17 @@ tests:
       TEST_TYPE: conformance-parallel
     workflow: openshift-e2e-powervs-ipi
   timeout: 6h0m0s
+- as: integration-tests-nodejoiner
+  run_if_changed: ^(cmd|pkg|data).*/(agent|imagebased|node-joiner)/
```
Comment (bfournie) [inline ci-operator/config/openshift/installer/openshift-installer-master.yaml:894]: Probably don't need imagebased here

**Pull request merged**